### ISSUE TITLE
Links with a  dot are not processed by turbo

### ISSUE THREAD

**Issue by silva96** [hotwired/turbo#385]
We have links in our platform that are user profiles like this:

domain.com/users/some.username

Whenever you click on those links, a normal request is made instead of a turbo request.

Is there any way to fix this? (I don't want to write a stimulus "link_controller#click" that calls Turbo.visit if it can be fixed from turbo side)

Thanks!

**Comment by terracatta** [comment]
If anyone else was reading this issue and asked themselves "are periods even allowed in the path?", [apparently they are](https://stackoverflow.com/questions/7555553/can-period-be-part-of-the-path-part-of-an-url).

**Comment by brunoprietog** [comment]
I have the same issue, it would be ideal if it could be solved directly from turbo.

**Comment by silva96** [comment]
> If anyone else was reading this issue and asked themselves "are periods even allowed in the path?", [apparently they are](https://stackoverflow.com/questions/7555553/can-period-be-part-of-the-path-part-of-an-url).

Yes, instagram does it all day https://www.instagram.com/ruby.tuesday/

**Comment by tleish** [comment]
TLDR: Turbo can handle periods, as long as the URL doesn't end in a way that suggests the URL is NOT an html page.

DETAILS:

Links with dots are processed with turbo:

- domain.com/users/some.username.html => fetched with turbo
- domain.com/users/some.username/ => fetched with turbo
- domain.com/users/some.png => NOT fetched with turbo
- domain.com/users/some.username => NOT fetched with turbo

Note in the instagram example, the URL ends in "/", indicating it's a directory

Turbo checks to see if a url is a directory or an html page

```javascript
function isHTML(url) {
  return !!getExtension(url).match(/^(?:|\.(?:htm|html|xhtml))$/)
}
```

Turbo skips non-html links (e.g. .json, .svg, .jpg, .png, etc).  

```javascript
  locationIsVisitable(location: URL) {
    return ... && isHTML(location)
  }
```

In the example of `domain.com/users/some.username`, how would you propose the javascript change to determine if the url is an html page vs a file with a `.username` extension?

**Comment by silva96** [comment]
Thanks, we solved it adding trailing_slash to the route.rb
